src/runtime/server/services/local/getServerSession.ts

```diff
@@ -0,0 +1,51 @@
+import { type H3Event, createError } from 'h3'
+import getURL from 'requrl'
+import { joinURL } from 'ufo'
+import { jsonPointerGet, useTypedBackendConfig } from '../../../helpers'
+import { getToken } from './getToken'
+
+// @ts-expect-error - #auth not defined
+import type { SessionData } from '#auth'
+import { useRuntimeConfig } from '#imports'
+
+function joinPathToApiURL(event: H3Event, path: string) {
+  const { origin, pathname, fullBaseUrl } = useRuntimeConfig().public.auth.computed
+
+  let baseURL
+  if (origin) {
+    // Case 1: An origin was supplied by the developer in the runtime-config. Use it by returning the already assembled full base url that contains it
+    baseURL = fullBaseUrl
+  }
+  else {
+    // Case 2: An origin was not supplied, we determine it from the request
```

Comment:
This probably needs to happen only when `trustHost` is set (look at `authjs` implementation)